### PR TITLE
ADD [WEB] > add urlProvider

### PR DESCRIPTION
**Pull request by jbbeuzelin**
- Enable custom apiPrefix
- Return an absolute path on `Url` property VS partial path before
- Abstract Url logic in UrlProvider

### CODE REVIEW

File: Web/RDD.Web/Serialization/UrlProvider.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            return DEFAULT_API_PREFIX;
+        }
+
+        public virtual string GetUrlTemplateFromEntityType(Type entityType)
+        {
+            var apiRadical = _pluralizationService.GetPlural(entityType.Name).ToLower();
+            var request = _httpContextAccessor.HttpContext.Request;
+
+            return $"{request.Scheme}://{request.Host.Value}/{GetApiPrefix()}/{apiRadical}/{{0}}";
```
Comment (nfaugout) [inline Web/RDD.Web/Serialization/UrlProvider.cs]: I wonder if this {{0}} is a good idea. It creates a strong dependency between that method and the callers (SerializeProperties for instance), that should assume thaht there will be a {0} inside the template.

We should maybe add a parameter inside that method => string GetEntityUrl(Type entityType, string entityId)

**Comment by rducom** [comment]
I have tried to achieve a routing prefix for default routes with this method : https://github.com/LuccaSA/RestDrivenDomain/blob/various.things/Web/RDD.Web/Helpers/RddRoutingExtensions.cs

But this doesn't applies to all routes... I think something must exists in standard for a such a feature, but I haven't take time to find it yet... The aspnet core routing middleware is extensible, a solution must exists.

File: Web/RDD.Web/Serialization/UrlProvider.cs (hunk 1 above)
Comment (jbbeuzelin) [reply Web/RDD.Web/Serialization/UrlProvider.cs]: i agree, i will try a quickwins

Comment (nfaugout) [reply Web/RDD.Web/Serialization/UrlProvider.cs]: @jbbeuzelin up, if you want we can handle this little refacto ;)

**Pull request merged**